ledger/controllers/surveyor.js

```diff
@@ -750,3 +751,38 @@ function getVoteRate () {
     })
   }
 }
+
+async function addSurveyorChoices (runtime, surveyor = {}) {
+  const payload = surveyor.payload || {}
+  surveyor.payload = payload
+  const adFree = payload.adFree || {}
+  payload.adFree = adFree
+  adFree.choices = await getAdjustedChoices(runtime, 'BAT', ['USD'])
+  return surveyor
+}
+
+async function getAdjustedChoices (runtime, base, currencies) {
+  const rates = await runtime.currency.rates(base, currencies)
+  return underscore.mapObject(rates, choicesPrices)
+}
+
+function choicesPrices (ratio) {
+  const table = [
+    [3, 5, 7, 10, 20],
+    [4, 6, 9, 12, 25],
+    [5, 8, 11, 17, 35],
+    [6, 10, 14, 20, 40],
+    [9, 12, 20, 35, 50],
+    [15, 25, 35, 50, 100],
+    [20, 35, 50, 85],
+    [30, 50, 70, 100]
+  ]
+  const priceIncrements = [1, 0.8, 0.6, 0.5, 0.35, 0.2, 0.15, 0.1]
+  let index = underscore.findIndex(priceIncrements, (increment) => {
+    return increment <= ratio
+  })
+  if (underscore.isUndefined(index)) {
+    index = priceIncrements.length - 1
+  }
+  return table[index]
+}
```

Comment:
I may have found a bug here, i'm getting that any ratio under `.1` will return undefined instead of the highest option.

```
> choicesPrices(.1)
[ 30, 50, 70, 100 ]
> choicesPrices(.09)
undefined
```

If this is a valid bug, would you write a test case?

Reply:
for sure